scripts/zones/Sauromugue_Champaign/npcs/qm3.lua

```diff
@@ -10,28 +10,27 @@ require("scripts/globals/quests")
 -----------------------------------
 
 function onTrade(player, npc, trade)
-    local grapplingCS = player:getCharVar("thickAsThievesGrapplingCS")
-
-    if grapplingCS >= 2 and grapplingCS <= 7 and npcUtil.tradeHas(trade, 17474) then
+    if player:getQuestStatus(WINDURST, tpz.quest.id.windurst.AS_THICK_AS_THIEVES) == QUEST_ACCEPTED and npcUtil.tradeHas(trade, 17474) then
         player:messageSpecial(ID.text.THF_AF_WALL_OFFSET + 3, 0, 17474) -- You cannot get a decent grip on the wall using the [Grapnel].
     end
 end
 
 function onTrigger(player, npc)
-    local thickAsThieves = player:getQuestStatus(WINDURST, tpz.quest.id.windurst.AS_THICK_AS_THIEVES)
-    local grapplingCS = player:getCharVar("thickAsThievesGrapplingCS")
-
-    if thickAsThieves == QUEST_ACCEPTED then
-        if grapplingCS == 3 then
-            local gob = GetMobByID(ID.mob.CLIMBPIX_HIGHRISE)
-
-            if not gob:isSpawned() then
-                player:messageSpecial(ID.text.THF_AF_MOB)
-                gob:setSpawn(414, 16, -131)
-                SpawnMob(ID.mob.CLIMBPIX_HIGHRISE):updateClaim(player)
+    if player:getQuestStatus(WINDURST, tpz.quest.id.windurst.AS_THICK_AS_THIEVES) == QUEST_ACCEPTED then
+        if not player:hasKeyItem(tpz.ki.FIRST_SIGNED_FORGED_ENVELOPE) then
```

Comment:
This code can be simplified:
```lua
if npc:getLocalVar("[QM]Select") == 1 and npcUtil.popFromQM(player, npc, ID.mob.CLIMBPIX_HIGHRISE, {radius = 1, hide = 0}) then
    player:messageSpecial(ID.text.THF_AF_MOB)
end
```

The popFromQM function will ...
* check if the mob's already spawned
* set the mob's spawn location to near the NPC (within radius of 1 yalm)
* spawn the mob
* claim the mob

Reply:
Thanks, I'll get this added.  This was another quest that I intended to go in and fix something simple, then the scope became much bigger.  I briefly skimmed the gambling portion of the quest; it can likely be improved.